### DISCUSSION TITLE
Implement round functionality for ADS integration

### DISCUSSION THREAD

**Discussion by gijbelsy** [home-assistant/feature-requests#1563] [Integration enhancements]
### Integration name

ADS

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/ads/#sensor

### Describe the enhancement

Implement a round function to be used for sensors.
In example:
```
if round_decimal is None:
      self._state_dict[state_key] = tempvalue
else:
      self._state_dict[state_key] = round(tempvalue, round_decimal)
```

### Use cases

The Beckhoff PLC stores sensordata several digits behind the decimal point.
Due to process fluctuations these values also fluctuate neglectably, but are recored by HA en therfore the system gets flooded.
A temperature sensor for example usually only holds 2 digits behind the decimal point.
Implementing a round function not only cleans up the interface, but also prevents state-change flooding.

**Comment by github-actions** [comment]
🏷️ I've automatically added the `integration: ads` label to help categorize this feature request.

**Comment by gjohansson-ST** [comment]
I guess the integration could suggest a nice precision but you can also yourself on each sensor select the precision in the entity settings (if the sensors are created properly being numerical ones).

**Comment by gijbelsy** [comment]
@gjohansson-ST True, and the integration creates the sensors properly to achieve this. However this still floods the HA backend, because the value is sent to HA before the rounding. Therefore a state change is triggered, and the value is recorded in the recorder & history database.

It would be even better to have 2 extra options:

- round
- maximum update interval in time or value change

With the maximum update interval a temperature sensor jumping from x.01 to x.02 and back gets filtered out.
All of this can be done inside the Beckhoff PLC, but that requires unnecessary extra code and variables inside the PLC program.
